fix: initialize message fields before applying hub fields

`Hub.Publish` can panic when the hub has default fields and the published message has a nil `Fields` map.

Current code:

```go
func (h *Hub) Publish(m Message) {
	for k, v := range h.fields {
		m.Fields[k] = v
	}

	for _, sub := range h.matcher.Lookup(m.Topic()) {
		sub.Set(m)
	}
}
```

If `h.fields` is non-empty and `m.Fields` is nil, this causes:

```text
panic: assignment to entry in nil map
```

Example:

```go
h := New().With(Fields{"source": "test"})
h.Publish(Message{Name: "log.debug"})
```

## Expected behavior

Publishing a message without explicit fields should work even when the hub has default fields.

## Suggested fix

Initialize `m.Fields` when needed:

```go
if len(h.fields) > 0 && m.Fields == nil {
	m.Fields = Fields{}
}
```

before copying hub fields into the message.

## Acceptance criteria

- Publishing with hub-level fields and nil message fields does not panic.
- Add a regression test.
- Existing tests continue to pass.

--- a/hub.go
+++ b/hub.go
@@ -24,6 +24,10 @@ func New() *Hub {
 
 // Publish will send an event to all the subscribers matching the event name.
 func (h *Hub) Publish(m Message) {
+	if len(h.fields) > 0 && m.Fields == nil {
+		m.Fields = Fields{}
+	}
+
 	for k, v := range h.fields {
 		m.Fields[k] = v
 	}

--- a/hub_test.go
+++ b/hub_test.go
@@ -195,6 +195,20 @@ func testPublishDoesNotPanicWhenSubscriberClosesAfterLookup(t *testing.T, subscr
 	}
 }
 
+func TestPublishInitializesNilMessageFieldsWhenHubHasFields(t *testing.T) {
+	h := New().With(Fields{"source": "test"})
+
+	sub := h.Subscribe(1, "log.*")
+	defer h.Unsubscribe(sub)
+
+	require.NotPanics(t, func() {
+		h.Publish(Message{Name: "log.debug"})
+	})
+
+	msg := <-sub.Receiver
+	require.Equal(t, Fields{"source": "test"}, msg.Fields)
+}
+
 func newMessageCounter(s Subscription) *messageCounter {
 	ms := &messageCounter{sub: s, c: 0}
 	go func(ms *messageCounter) {

